### ISSUE TITLE
Camera config 

### ISSUE THREAD

**Issue by tonhathuy** [PRBonn/SIMP#3]
Thank you, this project is great!
I have a few questions about camera pose config:
```"K" : [380.07025146484375, 0.0, 324.4729919433594, 0.0, 379.66119384765625, 237.78517150878906, 0.0, 0.0, 1.0],
  "cam_poses" : [0.1, 0, 0, 0, 0, -0.1, 0, -1.5707963267948966, -0.1, 0, 0, 3.141592653589793, 0, 0.1, 0,  1.5707963267948966],
  "angles" : [ 0, -1.5707963267948966,3.141592653589793, 1.5707963267948966],
  "min_z" : -1.55708286,
  "camera_z" : 0.63
```
As I understand it camera_pose is the camera position relative to base_link:
for each cam is (x,y,z,angle) with angle being the rotation angle of the z axis. According to your configuration, the 4 cameras rotate at 4 angles 0, -90 180 90. Because I'm only using 1 camera, I only customized the first camera
I have customized a few parameters to suit my camera (realsense d435i):
```
"K" : [609.7203979492188, 0.0, 328.297119140625, 0.0, 609.4514770507812, 243.18663024902344, 0.0, 0.0, 1.0],
  "cam_poses" : [0.2, 0, 0, 0, 0, -0.1, 0, -1.5707963267948966, -0.1, 0, 0, 3.141592653589793, 0, 0.1, 0,  1.5707963267948966],
  "angles" : [ 0, -1.5707963267948966,3.141592653589793, 1.5707963267948966],
  "min_z" : 0,
  "camera_z" : 2.0255000637941714
```
My camera is located about 2m above the floor and 0.2 x axis away from base_link
Also, because my camera doesn't have an IMU, I published a topic /camera_pose using the `TransformListener` function. I have configured this topic correctly with the camera position relative to /map.
And here is the result I got:
- the red arrow above is /camera_pose.
- In addition, I used the weights that you retrained for the 3D object detection model, but I almost couldn't detect any object correctly. It seems the model was overfit with your environment, so I used back to the original model.![image](https://github.com/PRBonn/SIMP/assets/45118883/ab8176be-f5eb-4c28-835f-96554c0b6bc5)
![image](https://github.com/PRBonn/SIMP/assets/45118883/8a3df0d2-6423-425e-865d-0db251f44f8f)

Could you tell me what causes the object's position to be wrong?
thanks

**Comment by FullMetalNicky** [comment]
For the localization, having a wide field of view is pretty crucial. We didn't test the localization for only one camera, so we can't even tell if it works.
These DL models are super sensitive to view-point changes. We refined the model on data from our robot - the camera was around 70cm above the ground, looking straight ahead. That's an entirely different view-point than what you have, e.g. we actually saw tables from below. 
You can try to use the original model provided by Meta, or refine on your own data. But the degraded performance is to be expected, at least from my experience.

**Comment by tonhathuy** [comment]
Hi Nicky, 
I tried using Meta's original model on my ROSbag and the one your rosbag provides but it seems that the simp.update() function doesn't work, it doesn't match any objects while I still see the detectable model.
If this is also caused by the camera's viewing angle, how did you re-train this model? . What tool did you use to label the data? Because I see that the data must be labeled with 8 points for the 3D object, so labeling is quite difficult.
Also, what configuration did you use to re-train this model? I see that this model needs up to 48 A100 GPUs for training in Meta's paper, a crazy number.
Thank you

**Comment by FullMetalNicky** [comment]
I don't remember the classes of the original model, but I would assume the reason it doesn't work, is because of class incompatibility. E.g. class 0 for us is a sink, but in the Meta model it's some other number. So perhaps you need to add a mapping between these two class arrays.  

Yes, labeling is quite difficult. Which is why we came up with an automatic way of extracting the labels - we explain in our paper how we do it in our lab, and there is also some code dealing with the conversion to the required input format in NewAutomaticLabeling.py. 

We fine-tuned on a single GPU - if you look carefully in the GitHub Readme for their code, they give the parameters for training on a single GPU. Our GPU also was not fancy, maybe a RTX 2080 or something like that. You can re-train with standard hardware, but you just need to be patient (for it was 2-3 days per training).
